cdap-app-fabric/src/main/java/co/cask/cdap/app/preview/PreviewServerModule.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.app.preview;
+
+import co.cask.cdap.api.dataset.module.DatasetDefinitionRegistry;
+import co.cask.cdap.common.conf.Constants;
+import co.cask.cdap.data.runtime.DataSetsModules;
+import co.cask.cdap.data2.datafabric.dataset.RemoteDatasetFramework;
+import co.cask.cdap.data2.dataset2.DatasetDefinitionRegistryFactory;
+import co.cask.cdap.data2.dataset2.DatasetFramework;
+import co.cask.cdap.data2.dataset2.DefaultDatasetDefinitionRegistry;
+import co.cask.cdap.gateway.handlers.CommonHandlers;
+import co.cask.cdap.gateway.handlers.preview.PreviewHttpHandler;
+import co.cask.http.HttpHandler;
+import com.google.inject.PrivateModule;
+import com.google.inject.Scopes;
+import com.google.inject.assistedinject.FactoryModuleBuilder;
+import com.google.inject.multibindings.Multibinder;
+import com.google.inject.name.Names;
+
+/**
+ * Provides bindings required create the {@link PreviewServer}.
+ */
+public class PreviewServerModule extends PrivateModule {
+  @Override
+  protected void configure() {
+    install(new FactoryModuleBuilder()
```

Comment:
But shouldn't they are the same since they always run in the same JVM (even in distributed mode)?? The only entry point to preview system is through http, right? (in unit-test, you can hit the preview manager directly, but that shouldn't affect how the guice modules are organized).